### ISSUE TITLE
Handle restricted posting permissions

### ISSUE THREAD

**Issue by auscompgeek** [matrix-org/matrix-appservice-slack#178]
It is possible to restrict posting in #general to either admins only or admins + selected users. There should be an easy way to map these to power levels to faithfully bridge such a channel.

**Comment by Half-Shot** [comment]
Yep, we'd want to look at https://api.slack.com/docs/permissions-api#change_management

**Comment by auscompgeek** [comment]
I'm not sure this API is relevant. I'm referring to this feature: https://get.slack.help/hc/en-us/articles/360004635551-Manage-channel-posting-permissions-

**Comment by auscompgeek** [comment]
This is probably mostly relevant for automatically provisioned rooms, since I've separated out #533 as a separate concern.
